### PR TITLE
A User cannot delete another user

### PR DESCRIPTION
**Pull request by JuanSaieh**
Changes:
 - A User cannot delete another User
 - Users can only modify themselves
 
Screen Shot 2019-08-12 at 5 47 50 PM

Trello Link: https://trello.com/c/Iqo0IKkl

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -20,9 +21,16 @@ def create
...
   def destroy
-    @user.destroy
-    redirect_to users_path
+    unless @allow_action
+      redirect_to users_path, alert: 'Permission Denied - Cannot delete another User'
+    else
+      @user.destroy
+      redirect_to users_path, notice: 'User deleted successfully'
+    end
+
```
Comment (Jucerweb) [inline app/controllers/users_controller.rb]: Remove this space

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -20,9 +21,16 @@ def create
     end
   end
 
+  def show; end
+
   def destroy
-    @user.destroy
-    redirect_to users_path
+    unless @allow_action
```
Comment (Jucerweb) [inline app/controllers/users_controller.rb]: You can write this way: 
```  
def destroy
    return  redirect_to users_path, alert: 'Permission Denied - Cannot delete another User' unless @allow_action
    @user.destroy
    redirect_to users_path, notice: 'User deleted successfully'
end
```

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (JuanSaieh) [reply app/controllers/users_controller.rb]: Done!

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (JuanSaieh) [reply app/controllers/users_controller.rb]: Thanks!

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -20,9 +21,12 @@ def create
     end
   end
 
+  def show; end
+
   def destroy
+    return  redirect_to users_path, alert: 'Permission Denied - Cannot delete another User' unless @allow_action
     @user.destroy
-    redirect_to users_path
+    redirect_to users_path, notice: 'User deleted successfully'
```
Comment (albertomejia92) [inline app/controllers/users_controller.rb]: If the user is not deleted, this should send the same message?
Please add a conditional to handle the messages in order to notify when an error occurred.

Comment (JuanSaieh) [reply app/controllers/users_controller.rb]: Devise is handling sessions!

**Pull request merged**